src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -301,11 +301,7 @@ private function parseXmlAttributes(\DOMNode $node)
         $data = array();
 
         foreach ($node->attributes as $attr) {
-            if (ctype_digit($attr->nodeValue)) {
-                $data['@'.$attr->nodeName] = (int) $attr->nodeValue;
-            } else {
-                $data['@'.$attr->nodeName] = $attr->nodeValue;
-            }
+            $data['@'.$attr->nodeName] = $attr->nodeValue;
```

Comment:
This is a BC break in case someone relies on the type of the decoded data (this could happen easily with PHP7 et strict types).

Maybe you could check if the cast to `int` loses information and keep a string only in this case.